### ISSUE TITLE
KioskBridge: Make available via app store or test drive

### ISSUE THREAD

**Issue by urapadmin** [arch-kiosk/arch-kiosk-office#1985]
KioskBridge is currently only available for Brown devices that we set up physically by connecting them to a certain computer.
For KHPP and MKAP that is not at all doable. So: The app store it is.

**Comment by urapadmin** [comment]
### KioskBridge 0.2.2
I have clicked a lot and finally Apple thinks it has invited you guys to test KioskBridge. Please take an iPad that does not already have KioskBridge and try to access the test version. It seems you'll have to get the TestDrive App on that iPad first.
Does it work?

**Comment by luizaogs** [comment]
Kidding, I figured it out. Had to add myself as a tester. Once Laurel accepts her invitation she will need to be added as well, I think.

**Comment by urapadmin** [comment]
aha. In App Connect? Interesting. I did not know you even have access to it. But I can confirm you are added to the test group and have installed KioskBridge. 
Should I have told you that as a server address you have to use the whole "https://test.arch-kiosk.brown.edu"?

**Comment by luizaogs** [comment]
Ha that part I figured out :)

**Comment by urapadmin** [comment]
Is that the test result :o)  ?

**Comment by luizaogs** [comment]
Ja.

**Comment by urapadmin** [comment]
Meanwhile we got KioskBridge approved by Apple for testing and so we can use a link to install it on a device:
(https://testflight.apple.com/join/Q81fJAvA)[https://testflight.apple.com/join/Q81fJAvA]
Does anybody have a device that has not been used already? @didi-elbehaedi You are welcome to test this, too if you have access to a device and to Luiza

**Comment by luizaogs** [comment]
If I uninstall the app I have and install it with the link does that work for testing purposes?

**Comment by luizaogs** [comment]
Hmm yeah I downloaded it again and it worked but I think it was already in my TestFlight so I don't think we can trust this test.

**Comment by urapadmin** [comment]
perhaps more if you delete TestFlight, too?

**Comment by urapadmin** [comment]
> Oh I deleted both. But then when I redownloaded TestFlight (as prompted by the link) Bridge was already in there.

then I agree that it isn't all that convincing. We need a device that isn't connected to anything, yet._

**Comment by didi-elbehaedi** [comment]
Sounds great! Will try it out and let you know how it goes.

**Comment by luizaogs** [comment]
I just ran through this with Didi and it all worked fine until she tried to upload it. We got this error: 
![340598410_1482426169317823_1891827962618479262_n](https://user-images.githubusercontent.com/67970402/231573809-8edd4166-f1bb-4e0d-b350-36893370e253.jpg)

We tried it twice and it failed twice. She followed the same exact steps that I did, and mine worked. So not sure what the issue is.

**Comment by luizaogs** [comment]
Oh. It might be because she lacks privileges?

**Comment by urapadmin** [comment]
what user is Didi using?

**Comment by luizaogs** [comment]
test, which initially had no privileges. I now gave it the field_worker and operator privileges so she can try again and she will do that soon.

**Comment by urapadmin** [comment]
ah. okay, that would make sense. The logs are not very clear about what kind of authorization went wrong but before I look into it I'll wait for Didi's next turn. She also should not have been allowed to download, though (and I saw that some download went wrong...), so I am a bit surprised about that part.

**Comment by luizaogs** [comment]
So the download worked the first time, and it worked again now that the privileges are there (to make sure they had been applied I asked Didi to reconnect Bridge). But the upload is still not working. She got the same error.

**Comment by urapadmin** [comment]
in that case, what is the iOS Version of the device?

**Comment by luizaogs** [comment]
16 something, I think. She had to update it to use Bridge so it's the latest update available.

**Comment by urapadmin** [comment]
hm. That should work. You were able to upload, right?

**Comment by luizaogs** [comment]
Yes.

**Comment by urapadmin** [comment]
I think I figured this out. The user "test" you guys were using has been used before for other tests, so the privileges were pretty messed up. That explains why user test was able to download without being applied to any group. 
Applying the group "field_worker" did not help because that group did not have upload privileges on Beset. There must have been some former test run about privileges that lead to this trouble.
I have now rectified the mess:
"test" is in group "field_worker" and "field_worker" has upload and download privileges.

In other words: Now it should work for Didi.

**Comment by luizaogs** [comment]
Ok, now she can’t even download. I tried logging in with the test user and I can’t even get to the hub so it seems like privileges are still messed up.

I just created an admin login for her so we can actually see if her Bridge is working. If we need to keep testing privileges for this I’ll do it myself.

**Comment by luizaogs** [comment]
With admin privileges in her account (deb, which I created) we were able to go through download and upload and it worked. So I think the purpose of this ticket is good to go - using the Apple link to download Bridge works just fine.

If you want to look into the privilege issue further it might make sense to open another ticket?

**Comment by urapadmin** [comment]
yes, I agree. This is really just a test issue then. I am sure the privileges work fine, I just missed something.
You may open a ticket for the PVD privileges and close this one.

**Comment by urapadmin** [comment]
added the download privilege for the test user.

**Comment by luizaogs** [comment]
I just tried it with the test user one last time and the whole thing worked, so privileges seem fine too and so I close this.
